File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> numbers) {
    vector<int> countSet(1001, 0);
    int sSize = 0;
    
    for (auto& n : numbers)
    {
        countSet[n]++;
        sSize += to_string(n).size();
    }

    if (countSet[0] == numbers.size())
        return "0";

    vector<string> nDict(1001);
    for (int i = 0; i <= 1000; i++)
    {
        nDict[i] = to_string(i);
    }
    sort(nDict.begin(), nDict.end(), [](const string& lhs, const string& rhs) -> bool
        {
            string a = lhs + rhs;
            string b = rhs + lhs;

            return a >= b;
        }
    );

    string answer;
    answer.reserve(sSize);
    for (auto& it : nDict)
    {
        int i = stoi(it);
        while (countSet[i]--)
        {
            answer += it;
        }
    }

    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cs
using System;
using System.Text;

public class Solution {
    public string solution(int[] numbers) {
        int[] countSet = new int[1001];
        foreach(var n in numbers)
        {
            countSet[n]++;
        }
        
        if (countSet[0] == numbers.Length)
            return "0";
        
        string[] nDict = new string[1001];
        
        for (int i = 0; i<= 1000; i++)
        {
            nDict[i] = Convert.ToString(i);
        }
        
        Array.Sort(nDict, (x, y) =>
            {
                string a = x + y;
                string b = y + x;
                return b.CompareTo(a);
            });
        
        string answer = "";
        StringBuilder stringBuilder = new StringBuilder();
        
        foreach(var it in nDict)
        {
            int i = Convert.ToInt32(it);
            while(countSet[i] > 0)
            {
                stringBuilder.Append(it);
                countSet[i]--;
            }
        }
        
        answer = stringBuilder.ToString();
        
        return answer;
    }
}